### DISCUSSION TITLE
with_expiry(Expiry::OnSessionEnd) does not seem to work

### DISCUSSION THREAD

**Discussion by iinov** [maxcountryman/tower-sessions#148] [Q&A]
Hello,

I try to use a `SessionManagerLayer` like this:

```
let session_layer = SessionManagerLayer::new(session_store)
           .with_name("my_session")
           .with_secure(false)
           .with_expiry(Expiry::OnSessionEnd);
```
Unfortunately, the the cookie expires in 2 weeks and is not a cookie of session. I have the same result with:

```
let session_layer = SessionManagerLayer::new(session_store)
           .with_name("my_session")
           .with_secure(false);
```

Did I miss something?

**Comment by maxcountryman** [comment]
Can you provide a minimal and complete application that demonstrates what you’re seeing?

**Reply by maxcountryman** [reply]
I think I see what's going on: in porting the Django implementation, we missed [the explicit check](https://github.com/django/django/blob/9c6d7b4a678b7bbc6a1a14420f686162ba9016f5/django/contrib/sessions/middleware.py#L48-L49) for this.

This should be addressed in `0.10.1`.

**Comment by iinov** [comment]
Thank you for your responsiveness!

So, I am waiting for release 0.10.1.

**Reply by maxcountryman** [reply]
0.10.1 is already released.

**Reply by iinov** [reply]
Thanks a lot.
